libraries/AP_Mount/AP_Mount_CADDX.h

```diff
@@ -0,0 +1,84 @@
+/*
+  CADDX mount using serial protocol backend class
+ */
+#pragma once
+
+#include "AP_Mount_Backend_Serial.h"
+
+#if HAL_MOUNT_CADDX_ENABLED
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_Common/AP_Common.h>
+
+#define AP_MOUNT_CADDX_RESEND_MS   1000    // resend angle targets to gimbal once per second
+
+#define AXIS_MAX 2047
+#define AXIS_MIN -2048
+
+typedef enum {
+    GIMBAL_MODE_FOLLOW = 0,
+    GIMBAL_MODE_TILT_LOCK = (1<<0),
+    GIMBAL_MODE_ROLL_LOCK = (1<<1),
+    GIMBAL_MODE_PAN_LOCK  = (1<<2),
```

Comment:
if possible let's use "Yaw" instead of "Pan".  I think we should avoid synonyms where we can and I think I've gotten rid of most of them

Reply:
I think this overlaps with the Siyi mount's "FPV" mode which we've had users asking for for quite some time.  Siyi's FPV mode means all the angle targets are in vehicle body frame.

Siyi's FPV mode is only valid while the mount is in RC_TARGETING mode so if this mount is the same we need to be sure this is enfoced somehow.  We don't want the user's aux switch leading to the mount not working while (for example) it tries to point at a POI.

I wonder how much flexibility the CADDX supports.  For example can roll be locked (e.g. earth frame) while pitch is not (e.g. body-frame)?  What would happen if the vehicle rolled 90 degrees (e.g. knife edge)?

re naming, I think we will probably need to replace the existing "yaw lock" boolean with a "lock_mode" that includes the new options.  this should probably be visible to all gimbals even if they don't all support it.  E.g. this enum should be moved up to AP_Mount.

Reply:
we will need to discuss